client/lib/lib.rs

```diff
@@ -0,0 +1,612 @@
+/// it is incompatible with `maybe_target_purse`.
+/// * `deploy` - Deploy arguments container struct holding options for this deploy.
+#[allow(clippy::too_many_arguments)]
+pub fn transfer(
+    maybe_rpc_id: &str,
+    node_address: &str,
+    verbose: bool,
+    amount: &str,
+    maybe_source_purse: &str,
+    maybe_target_purse: &str,
+    maybe_target_account: &str,
+    deploy: DeployStrParams<'_>,
+) -> Result<JsonRpc> {
+    let target = parsing::get_transfer_target(maybe_target_account, maybe_target_purse)?;
+
+    // transfer always invoke the standard payment
+    let payment = parsing::parse_standard_payment_info(amount)?;
+    let amount = U512::from_dec_str(amount)
+        .map_err(|err| Error::FailedToParseUint(UIntParseError::FromDecStr(err)))?;
+
+    let source_purse = parsing::purse::parse(maybe_source_purse).ok();
+
+    RpcCall::new(maybe_rpc_id, node_address, verbose)?.transfer(
+        amount,
+        source_purse,
+        target,
+        deploy.try_into()?,
+        payment,
+    )
+}
+
+/// Signs a deploy file.
+///
+/// * `input_path` specifies the path to the deploy file.
+/// * `secret_key` specifies the key with which to sign the deploy.
+/// * `maybe_output` specifies the output file, or if left empty, will log it to `stdout`.
+pub fn sign_deploy_file(input: &str, secret_key: &str, maybe_output: &str) -> Result<()> {
+    let input_path = parsing::input::get(input);
+    let secret_key = parsing::secret_key::parse(secret_key)?;
+    let maybe_output = parsing::output::get(maybe_output);
+    Deploy::sign_deploy_file(&input_path, secret_key, maybe_output)
+}
+
+/// Container for `Deploy` construction options.
+pub struct DeployStrParams<'a> {
+    /// Formatted string of `SecretKey`.
+    pub secret_key: &'a str,
+    /// Formatted string of `Timestamp`.
+    pub timestamp: &'a str,
+    /// Formatted string of `ttl`.
+    pub ttl: &'a str,
+    /// List of formatted `DeployHash` strings.
+    pub dependencies: &'a [&'a str],
+    /// Gas price (uint) as a string.
+    pub gas_price: &'a str,
+    /// Chain name.
+    pub chain_name: &'a str,
+}
+
+impl<'a> TryInto<DeployParams> for DeployStrParams<'a> {
+    type Error = Error;
+
+    fn try_into(self) -> Result<DeployParams> {
+        let DeployStrParams {
+            secret_key,
+            timestamp,
+            ttl,
+            gas_price,
+            dependencies,
+            chain_name,
+        } = self;
+        parsing::parse_deploy_params(
+            secret_key,
+            timestamp,
+            ttl,
+            gas_price,
+            dependencies,
+            chain_name,
+        )
+    }
+}
+
+/// Container for payment related arguments.
+#[derive(Default)]
+pub struct PaymentStrParams<'a> {
+    /// Payment amount - all other `payment_*` arguments are ignored and the
+    ///   standard payment contract is used if this is set.
+    pub payment_amount: &'a str,
```

Comment:
Since we have various constructors for this struct, I think if we make all the member variables private we'll be making it much harder to misuse.  Same comment for the `SessionStrParams`.

Reply:
Got it, thanks.